gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/language/grammar/StructureElementVisitor.java

```diff
@@ -32,7 +32,11 @@ public StructureElementVisitor(final GremlinAntlrToJava antlr) {
 
     @Override
     public Vertex visitStructureVertex(final GremlinParser.StructureVertexContext ctx) {
-        return new ReferenceVertex(antlr.argumentVisitor.visitGenericLiteralArgument(ctx.genericLiteralArgument()),
-                (String) antlr.argumentVisitor.visitStringArgument(ctx.stringArgument()));
+        if (ctx.getChildCount() == 4) {
```

Comment:
Nit: Prefer to use `if (ctx.stringArgument() == null)` instead of checking the child count.